Formula/spdlog.rb

```diff
@@ -32,20 +38,19 @@ def install
       #endif
     EOS
 
-    mkdir "spdlog-build" do
-      args = std_cmake_args + %W[
-        -Dpkg_config_libdir=#{lib}
-        -DSPDLOG_BUILD_BENCH=OFF
-        -DSPDLOG_BUILD_TESTS=OFF
-        -DSPDLOG_FMT_EXTERNAL=ON
-      ]
-      system "cmake", "..", "-DSPDLOG_BUILD_SHARED=ON", *args
-      system "make", "install"
-      system "make", "clean"
-      system "cmake", "..", "-DSPDLOG_BUILD_SHARED=OFF", *args
-      system "make"
-      lib.install "libspdlog.a"
-    end
+    args = std_cmake_args + %W[
+      -Dpkg_config_libdir=#{lib}
+      -DSPDLOG_BUILD_BENCH=OFF
+      -DSPDLOG_BUILD_TESTS=OFF
+      -DSPDLOG_FMT_EXTERNAL=ON
+    ]
+    system "cmake", "-S", ".", "-B", "build", "-DSPDLOG_BUILD_SHARED=ON", *args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+    system "cmake", "--build", "build", "--target", "clean"
+    system "cmake", "-S", ".", "-B", "build", "-DSPDLOG_BUILD_SHARED=OFF", *args
+    system "cmake", "--build", "build"
+    lib.install "build/libspdlog.a"
```

Comment:
The `clean` target will destroy some things that we normally keep as logs, so let's skip that.
```suggestion
    system "cmake", "-S", ".", "-B", "build-static", "-DSPDLOG_BUILD_SHARED=OFF", *args
    system "cmake", "--build", "build-static"
    lib.install "build-static/libspdlog.a"
```